sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_cache_utils.py

```diff
@@ -156,15 +203,22 @@ def _get_on_disk_hash_for_component(component: Component, in_memory_hash: str) -
             object_hash.update(content_hash.encode("utf-8"))
             return _CODE_INVOLVED_PREFIX + object_hash.hexdigest()
 
-    @staticmethod
-    def get_on_disk_cache_base_dir() -> Path:
+    @property
+    def _on_disk_cache_dir(self) -> Path:
         """Get the base path for on disk cache."""
         from azure.ai.ml._version import VERSION
-        return Path(tempfile.gettempdir()).joinpath(".azureml", "azure-ai-ml", VERSION, "cache", "components")
+        return Path(tempfile.gettempdir()).joinpath(
+            ".azureml",
+            "azure-ai-ml",
+            VERSION,
+            "cache",
+            self._client_hash,
+            "components",
+        )
 
     def _get_on_disk_cache_path(self, on_disk_hash: str) -> Path:
```

Comment:
If personal A created the disk cache, is it possible that personal B on the same machine use it? will permission error or something else happen?

Reply:
you mean 2 users submit pipelines at the same time and caused uncontrollable concurrent submission? It should be safe as there is no deleting operation in usual code (maybe I should move the clear function to `conftest.py`)

Reply:
I mean the former one submit pipeline at first, component id has been saved, however, the latter person has no access to the cached file generated by the first user, is that possible? We met this issue in sdk 1.5 I remember.

Reply:
Updated